x-pack/plugins/security_solution/public/common/components/hover_actions/use_hover_action_items.tsx

```diff
@@ -0,0 +1,310 @@
+  draggableId?: DraggableId;
+  enableOverflowButton?: boolean;
+  field: string;
+  handleHoverActionClicked: () => void;
+  isObjectArray: boolean;
+  isOverflowPopoverOpen?: boolean;
+  itemsToShow?: number;
+  onFilterAdded?: () => void;
+  onOverflowButtonClick?: () => void;
+  ownFocus: boolean;
+  showTopN: boolean;
+  stKeyboardEvent: React.KeyboardEvent<Element> | undefined;
+  timelineId?: string | null;
+  toggleColumn?: (column: ColumnHeaderOptions) => void;
+  toggleTopN: () => void;
+  values?: string[] | string | null;
+}
+
+interface UseHoverActionItems {
+  overflowActionItems: JSX.Element[];
+  allActionItems: JSX.Element[];
+}
+
+export const useHoverActionItems = ({
+  dataProvider,
+  dataType,
+  defaultFocusedButtonRef,
+  draggableId,
+  enableOverflowButton,
+  field,
+  handleHoverActionClicked,
+  isObjectArray,
+  isOverflowPopoverOpen,
+  itemsToShow = 2,
+  onFilterAdded,
+  onOverflowButtonClick,
+  ownFocus,
+  showTopN,
+  stKeyboardEvent,
+  timelineId,
+  toggleColumn,
+  toggleTopN,
+  values,
+}: UseHoverActionItemsProps): UseHoverActionItems => {
+  const kibana = useKibana();
+  const { timelines } = kibana.services;
+  // Common actions used by the alert table and alert flyout
+  const {
+    getAddToTimelineButton,
+    getColumnToggleButton,
+    getCopyButton,
+    getFilterForValueButton,
+    getFilterOutValueButton,
+    getOverflowButton,
+  } = timelines.getHoverActions();
+
+  const filterManagerBackup = useMemo(() => kibana.services.data.query.filterManager, [
+    kibana.services.data.query.filterManager,
+  ]);
+  const getManageTimeline = useMemo(() => timelineSelectors.getManageTimelineById(), []);
+  const { filterManager: activeFilterMananager } = useDeepEqualSelector((state) =>
+    getManageTimeline(state, timelineId ?? '')
+  );
+  const filterManager = useMemo(
+    () => (timelineId === TimelineId.active ? activeFilterMananager : filterManagerBackup),
+    [timelineId, activeFilterMananager, filterManagerBackup]
+  );
+
+  //  Regarding data from useManageTimeline:
+  //  * `indexToAdd`, which enables the alerts index to be appended to
+  //    the `indexPattern` returned by `useWithSource`, may only be populated when
+  //    this component is rendered in the context of the active timeline. This
+  //    behavior enables the 'All events' view by appending the alerts index
+  //    to the index pattern.
+  const activeScope: SourcererScopeName =
+    timelineId === TimelineId.active
+      ? SourcererScopeName.timeline
+      : timelineId != null &&
+        [TimelineId.detectionsPage, TimelineId.detectionsRulesDetailsPage].includes(
+          timelineId as TimelineId
+        )
+      ? SourcererScopeName.detections
+      : SourcererScopeName.default;
+  const { browserFields } = useSourcererScope(activeScope);
+
+  const showFilters =
+    values != null && (enableOverflowButton || (!showTopN && !enableOverflowButton));
```

Comment:
Can you put a comment here explaining the relationship between `showTopN and enableOverflowButton`? Just finding it hard to follow exactly why `topN` is chosen over the other actions

Reply:
Good point! Here is the comment I put, please let me know if there's better way to put it:
- In the case of `DisableOverflowButton`, we show filters only when topN is NOT opened. As after topN button is clicked, the chart panel replace current hover actions in the hover actions' popover, so we have to hide all the actions.
- in the case of `EnableOverflowButton`, we only need to hide all the items in the overflow popover as the chart's panel opens in the overflow popover, so non-overflowed actions are not affected.

Reply:
Awesome, thank you!